model/PermissionsService.php

```diff
@@ -197,9 +198,10 @@ private function validateResources(array $resultPermissions): void
     /**
      * @param array $addRemove
      * @param string $resourceId
+     * @param bool $isRecursive
      */
 
-    private function triggerEvents(array $addRemove, string $resourceId): void
+    private function triggerEvents(array $addRemove, string $resourceId, bool $isRecursive): void
```

Comment:
Is this the only call for triggerEvents? Not even from other extensions?

Reply:
Yes, I think that you did not see it, but the method is private.